code/game/machinery/iv_drip.dm

```diff
@@ -71,167 +26,45 @@
 	if(!ishuman(usr) || !iscarbon(target))
 		return
 
-	if(attached)
-		visible_message("<span class='warning'>[attached] is detached from [src].</span>")
-		attached = null
-		update_icon()
-		return
-
-	if(!target.dna)
-		to_chat(usr, "<span class='danger'>The drip beeps: Warning, incompatible creature!</span>")
-		return
-
 	if(Adjacent(target) && usr.Adjacent(target))
-		if(beaker)
-			usr.visible_message("<span class='warning'>[usr] attaches [src] to [target].</span>", "<span class='notice'>You attach [src] to [target].</span>")
-			attached = target
-			START_PROCESSING(SSmachines, src)
-			update_icon()
-		else
-			to_chat(usr, "<span class='warning'>There's nothing attached to the IV drip!</span>")
+		bag.afterattack(target, usr, 1)
 
+/obj/machinery/iv_drip/attack_hand(mob/user)
+	if(bag)
+		user.put_in_hands(bag)
+	bag.update_icon()

```

Comment:
if there's no bag, this is going to runtime.

Were the last 3 lines of this proc supposed to be indented one more?